### ISSUE TITLE
no_std: Add support for deprecated LayoutErr

### ISSUE THREAD

**Issue by mamins1376** [lovasoa/custom_error#28]
Been recently forging a small library, and decided to use this library but along the way got to see this:
```
error[E0412]: cannot find type `LayoutError` in module `core::alloc`
   --> /home/amin/.cache/cargo/registry/src/github.com-1ecc6299db9ec823/custom_error-1.9.1/src/error.rs:40:29
    |
40  | impl Error for core::alloc::LayoutError {}
    |                             ^^^^^^^^^^^ help: a struct with a similar name exists: `LayoutErr`
    |
   ::: /home/amin/.local/share/rustup/xtensa/rustc/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:399:1
    |
399 | pub struct LayoutErr {
    | -------------------- similarly named struct `LayoutErr` defined here

error: aborting due to previous error
```

The direct cause of this seems to be the fact that I currently am using a custom rustc, stuck on `1.48.0`, and [this deprecation](https://github.com/rust-lang/rust/blob/master/library/core/src/alloc/layout.rs#L401). Would really appreciate your consideration if possible to add support for this, thanks.

**Comment by lovasoa** [comment]
Hello and thanks for the report ! I see you are using the recently introduced nostd feature, is that on purpose ?
CI runs on rust 1.36 with both std and nostd mode and works fine. Could you open a PR that reproduces that on our CI ?

**Comment by lovasoa** [comment]
Okay, there was a bug in our ci. That's fixed, and compatibility with rust 1.36 is back.
